### PR TITLE
nrf54l15dk: patches needed for run MUBoot based dfu

### PR DESCRIPTION
**Pull request by nvlsianpu**
This introduce changes missung in nRF54l15DK support,
mainly follow what have been done for the PDK.

These patches are needed for run MUBoot based dfu.

### CODE REVIEW

**Review by nordicjm — APPROVED** [review]

File: modules/trusted-firmware-m/Kconfig.tfm.pm [hunk 1]
```diff
@@ -28,14 +28,14 @@ config PM_PARTITION_SIZE_TFM
 	default 0xFE00 if TFM_PROFILE_TYPE_MINIMAL && TFM_CMAKE_BUILD_TYPE_DEBUG && \
 		BOOTLOADER_MCUBOOT
 	default 0x10000 if TFM_PROFILE_TYPE_MINIMAL && TFM_CMAKE_BUILD_TYPE_DEBUG
-	default 0x6800 if TFM_PROFILE_TYPE_MINIMAL && BOOTLOADER_MCUBOOT && SOC_NRF54L15_ENGA_CPUAPP
+	default 0x6800 if TFM_PROFILE_TYPE_MINIMAL && BOOTLOADER_MCUBOOT && (SOC_NRF54L15_ENGA_CPUAPP || SOC_NRF54L15_ENGA_CPUAPP)
```
Comment (LuDuda) [inline modules/trusted-firmware-m/Kconfig.tfm.pm:31]: Did you mean:

```suggestion
	default 0x6800 if TFM_PROFILE_TYPE_MINIMAL && BOOTLOADER_MCUBOOT && (SOC_NRF54L15_ENGA_CPUAPP || SOC_NRF54L15_CPUAPP)
```

Same below?

**Review by LuDuda — COMMENTED** [review]
Thank you 🚀

File: subsys/partition_manager/pm.yml.settings [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # In nRF54L15 we place the TF-M non-secure storage partitions after the
 # TF-M non-secure application to avoid splitting the secure/non-secure
 # partitions more than necessary.
-#ifdef CONFIG_SOC_NRF54L15_ENGA_CPUAPP
+#ifdef (CONFIG_SOC_NRF54L15_ENGA_CPUAPP || CONFIG_SOC_NRF54L15_CPUAPP)
```
Comment (ArekBalysNordic) [inline subsys/partition_manager/pm.yml.settings:6]:
```suggestion
#if defined(CONFIG_SOC_NRF54L15_ENGA_CPUAPP) || defined(CONFIG_SOC_NRF54L15_CPUAPP)
```

Comment (ArekBalysNordic) [reply subsys/partition_manager/pm.yml.settings:6]: And the same below.

**Review by gchwier — APPROVED** [review]

File: subsys/partition_manager/pm.yml.settings [hunk 3]
```diff
@@ -3,7 +3,7 @@
 # In nRF54L15 we place the TF-M non-secure storage partitions after the
 # TF-M non-secure application to avoid splitting the secure/non-secure
 # partitions more than necessary.
-#ifdef CONFIG_SOC_NRF54L15_ENGA_CPUAPP
+#if defined (CONFIG_SOC_NRF54L15_ENGA_CPUAPP || CONFIG_SOC_NRF54L15_CPUAPP)
```
Comment (gchwier) [inline subsys/partition_manager/pm.yml.settings:6]: failed in CI:
```
/jenkins_workspace/workspace/latest_sdk-nrf_PR-17334/nrf/subsys/partition_manager/pm.yml.settings:6:46: error: missing ')' after "defined"
    6 | #if defined (CONFIG_SOC_NRF54L15_ENGA_CPUAPP || CONFIG_SOC_NRF54L15_CPUAPP)
```

**Review by gchwier — CHANGES_REQUESTED** [review]

**Review by gchwier — APPROVED** [review]

**Comment by kapi-no** [comment]
I think that the MCUboot configuration misses one entry to disable FPROTECT

```
CONFIG_FPROTECT=n
```

When building the Bluetooth Peripheral LBS sample with the following command:

```
west build -b nrf54l15dk/nrf54l15/cpuapp -- -DS
B_CONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y
```

I get the following output:

```
*** Booting MCUboot v2.1.0-dev-a41abbcbba5f ***
*** Using nRF Connect SDK v2.7.99-9b20dc0ee39b ***
*** Using Zephyr OS v3.6.99-9824af7b5a01 ***
I: Starting bootloader
I: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: none
I: Image index: 0, Swap type: none
I: Bootloader chainload address offset: 0xc000
I: Jumping to the first image slot
E: Protect mcuboot flash failed, cancel startup.

```

**Comment by kapi-no** [comment]
The last update fixed the issue with FPROTECT. The happy path for the swap mode works as expected:

```
*** Booting MCUboot v2.1.0-dev-5eba7ab4cbfc ***
*** Using nRF Connect SDK v2.7.99-8d9cb62c36df ***
*** Using Zephyr OS v3.6.99-25e278581141 ***
I: Starting bootloader
I: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: none
I: Image index: 0, Swap type: none
I: Bootloader chainload address offset: 0xc000
I: Jumping to the first image slot
*** Booting My Application v2.7.99-8d9cb62c36df ***
*** Using nRF Connect SDK v2.7.99-8d9cb62c36df ***
*** Using Zephyr OS v3.6.99-25e278581141 ***
Starting Bluetooth Peripheral LBS example
I: 2 Sectors of 4096 bytes
I: alloc wra: 0, fa0
I: data wra: 0, 30
I: SoftDevice Controller build revision: 
I: 6c 48 af c9 da 87 c6 b7 |lH......
I: 89 37 8d 4c 36 5f bd 63 |.7.L6_.c
I: b9 bf 7e c2             |..~.    
I: HW Platform: Nordic Semiconductor (0x0002)
I: HW Variant: nRF54Lx (0x0005)
I: Firmware: Standard Bluetooth controller (0x00) Version 108.44872 Build 3330792137
I: No ID address. App must call settings_load()
Bluetooth initialized
I: Identity: C0:22:B6:54:D0:2D (random)
I: HCI: version 5.4 (0x0d) revision 0x3245, manufacturer 0x0059
I: LMP: version 5.4 (0x0d) subver 0x3245
Advertising successfully started
Connected
I: Image index: 0, Swap type: none
I: Image index: 0, Swap type: none
I: Image index: 0, Swap type: none
I: Image index: 0, Swap type: none
I: Image index: 0, Swap type: test
*** Booting MCUboot v2.1.0-dev-5eba7ab4cbfc ***
*** Using nRF Connect SDK v2.7.99-8d9cb62c36df ***
*** Using Zephyr OS v3.6.99-25e278581141 ***
I: Starting bootloader
I: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Secondary image: magic=good, swap_type=0x2, copy_done=0x3, image_ok=0x3
I: Boot source: none
I: Image index: 0, Swap type: test
I: Starting swap using move algorithm.
I: Bootloader chainload address offset: 0xc000
I: Jumping to the first image slot
*** Booting My Application v4.7.99-8d9cb62c36df ***
*** Using nRF Connect SDK v2.7.99-8d9cb62c36df ***
*** Using Zephyr OS v3.6.99-25e278581141 ***
Starting Bluetooth Peripheral LBS example
I: 2 Sectors of 4096 bytes
I: alloc wra: 0, fd0
I: data wra: 0, 0
I: SoftDevice Controller build revision: 
I: 6c 48 af c9 da 87 c6 b7 |lH......
I: 89 37 8d 4c 36 5f bd 63 |.7.L6_.c
I: b9 bf 7e c2             |..~.    
I: HW Platform: Nordic Semiconductor (0x0002)
I: HW Variant: nRF54Lx (0x0005)
I: Firmware: Standard Bluetooth controller (0x00) Version 108.44872 Build 3330792137
I: No ID address. App must call settings_load()
Bluetooth initialized
I: Identity: C0:22:B6:54:D0:2D (random)
I: HCI: version 5.4 (0x0d) revision 0x3245, manufacturer 0x0059
I: LMP: version 5.4 (0x0d) subver 0x3245
Advertising successfully started
```

**Review by kapi-no — APPROVED** [review]

**Comment by nvlsianpu** [comment]
test-fw-nrfconnect-proprietary_esb - passed after re-run

**Review by tomi-font — APPROVED** [review]

**Comment by nordicjm** [comment]
> test-fw-nrfconnect-proprietary_esb - passed after re-run test-fw-nrfconnect-nrf-iot_nrf_provisioning - failure irrelevant test-fw-nrfconnect-ble_mesh - failure irrelevant (unstable) test-low-leve - failure irrelevant
> 
> The rest CIs are green

When re-running CI, go to the PR ci job itself (not individual jobs) then click build, the `skip passed` checkbox should be ticked, click run, it will only rerun failed CI but will update the status here and allow merging

**Review by ArekBalysNordic — APPROVED** [review]

**Comment by kapi-no** [comment]
@nvlsianpu, did you forget to bump the revision of sdk-zephyr in west manifest? It points still to the merged Zephyr PR?

**Comment by nvlsianpu** [comment]
@kapi-no will fix in a moment - did locally, forgot to push...

**Review by kapi-no — APPROVED** [review]

**Pull request merged**